### PR TITLE
ThingHandlerService apidoc

### PR DESCRIPTION
**Pull request by davidgraeff**
Add a usage example for getServices()

Signed-off-by: David Gräff <david.graeff@web.de>

### CODE REVIEW

File: bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java [hunk 1]
```diff
@@ -13,11 +13,25 @@
...
 
 /**
+ * The {@link ThingHandler#getServices()} function allows you to return a list of classes,
+ * implementing this interface, that will be started as OSGi services. 
+ * <p>
+ * Use this for example to start thing specific discovery services (extending {@link AbstractDiscoveryService}).
+ * Example:
+ * <pre>
+ * @NonNullByDefault
+ * public class MyThingDiscovery extends AbstractDiscoveryService implements ThingHandlerService {
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java]: We have to slightly change this as the important thing is to implement the interface which has to be registered as service:

```suggestion
 * public class MyThingDiscovery extends AbstractDiscoveryService implements DiscoveryService, ThingHandlerService {
```

File: bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java [hunk 2]
```diff
@@ -13,11 +13,25 @@
...
+ *     private @Nullable ThingHandler thingHandler; // A reference to your thing handler
  *
- * Interface for a service that provides access to a {@link ThingHandler}
+ *     public void activate() {
+ *         super(Collections.emptyMap()); // start AbstractDiscoveryService
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java]: I guess you want to call the super method. And passing `null` should be sufficient.

```suggestion
 *         super.activate(null); // start AbstractDiscoveryService
```

File: bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java [hunk 3]
```diff
@@ -13,11 +13,25 @@
...
+ *     }
+ *     public void setThingHandler(ThingHandler handler) { this.thingHandler = handler; }
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java]: I am not sure if we should add the `@Override` to all three methods. The IDE takes care of that automatically if you safe the file.

```suggestion
 *     @Override
 *     public void setThingHandler(ThingHandler handler) { this.thingHandler = handler; }
```

File: bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java [hunk 4]
```diff
@@ -13,11 +13,25 @@
...
+ *     public ThingHandler getThingHandler() { return this.thingHandler; }
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java]:
```suggestion
 *     public @Nullable ThingHandler getThingHandler() { return this.thingHandler; }
```

**Comment by davidgraeff** [comment]
`getServices()` was not meant to be used for thing handler bound discovery services. The underlying implementation would need to be changed. Closing for now.

**Pull request closed without merging**